lib/resty/upstream/healthcheck.lua

```diff
@@ -650,6 +729,10 @@ function _M.spawn_checker(opts)
 
     update_upstream_checker_status(u, true)
 
+    if dedup == true then
+        copy_status(ctx)
```

Comment:
This is not doing what you think - it will only call `copy_status` on the first iteration, and not for subsequent health checks.